src/config/configurator.ts

```diff
@@ -115,8 +115,7 @@ class ConfiguratorAspect implements cdk.IAspect {
     if (this.resourceType && cdk.CfnResource.isCfnResource(node) && node.cfnResourceType == this.resourceType) {
       node.addPropertyOverride(this.propertyName, this.propertyValue);
     }
-
-    if (this.resourceId && cdk.CfnResource.isCfnResource(node)) {
+    if (this.resourceId && cdk.CfnResource.isCfnResource(node) && node.node.path.includes(this.resourceId)) {
```

Comment:
The problem here is that you might catch multiple resources this way. For example, let's say there's `MyQueue` and `MyQueue2`. `MyQueue` is a substring of both of them, so the configurator would modify both of them.

Another huge problem is that when CDK constructs are nested, prefixes of parent resource IDs will get appended to the name of each resource. However, the length of logical IDs is limited, so if the prefix is too long, the resource ID may not even be part of thew logical ID.

For example, if `MyQueue` is nested inside, `L3NestedConstructWithAReallyReallyReallyLongName`, the logical ID of the queue might be something like `L3NestedConstructWithARea8K3D1H`. In this case, `the configurator will not be able to modify a property for `MyQueue`.

Reply:
That's a good point. Perhaps `path` has it. A unit tests for this would definitely be a great thing though.

For the subpath issue, I think checking for a substring as you noted has two potential problems:
1. I'm not sure if the path will always end with a `/`. I could see that not being the case for `L1` resources, which we use for Glue crawlers.
2. We need to make sure that the beginning of the substring is bounded as well (like `includes("/${this.resourceId}/")`.

We could solve both of those problems by using a regex though. The beginning of the substring would be bounded by either a `/` or the string start token (`^` in regex), and the end would be bound by either `/` or the string end token (`$` in regex)